### PR TITLE
[STORM-3025] Optimize Cluster methods with Caching to avoid loopy loops 

### PR DESCRIPTION
**Pull request by kishorvpatil**
Optimizing the `Cluster` methods that are called from with high frequency calls to speed-up scheduling time on new topologies on large clusters.

### CODE REVIEW

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java [hunk 1]
```diff
@@ -187,4 +187,29 @@ public double getTotalCpu() {
     public NormalizedResources getNormalizedResources() {
         return this.normalizedResources;
     }
+
+    public void removeOffHeap(final double offHeap) {
```
Comment (agresch) [inline storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: where is this used?  Why increment?

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java [hunk 2]
```diff
@@ -187,4 +187,29 @@ public double getTotalCpu() {
...
+        this.offHeap += offHeap;
+    }
+
+    public void remove(WorkerResources value) {
```
Comment (agresch) [inline storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: were these used?

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java (hunk 1 above)
Comment (kishorvpatil) [reply storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: This is unused method. I was trying some thing else. Let me delete this method.

File: storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java (hunk 2 above)
Comment (kishorvpatil) [reply storm-server/src/main/java/org/apache/storm/scheduler/resource/normalization/NormalizedResourceRequest.java]: Not used

**Comment by kishorvpatil** [comment]
@agresch I have addressed the code review comments.

**Pull request merged**